pkg/proc/native/proc_freebsd.go

```diff
@@ -360,19 +409,70 @@ func (dbp *nativeProcess) stop(cctx *proc.ContinueOnceContext, trapthread *nativ
 	if dbp.exited {
 		return nil, proc.ErrProcessExited{Pid: dbp.pid}
 	}
-	// set breakpoints on all threads
-	for _, th := range dbp.threads {
-		if th.CurrentBreakpoint.Breakpoint == nil {
-			if err := th.SetCurrentBreakpoint(true); err != nil {
-				return nil, err
+
+	var err error
+	dbp.execPtraceFunc(func() {
+		for _, th := range dbp.threads {
+			err = ptraceSuspend(th.ID)
+			if err != nil {
+				return
+			}
+		}
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if trapthread.Status == nil || (*sys.WaitStatus)(trapthread.Status).StopSignal() != sys.SIGTRAP {
+		return trapthread, nil
+	}
+
+	return dbp.stop2(trapthread.ID)
+}
+
+func (dbp *nativeProcess) stop2(tids ...int) (*nativeThread, error) {
```

Comment:
nit: I think we can improve this name a bit. Perhaps something like `postStop` or similar? We don't really stop anything here, we only really acknowledge breakpoints that have been hit.